dfencoder/scalers.py

```diff
@@ -48,6 +44,48 @@ def fit_transform(self, x: torch.Tensor):
         self.fit(x)
         return self.transform(x)
 
+class ModifiedScaler(object):
+    """Implements scaling using modified z score."""
+
+    def __init__(self):
+        self.median = None
+        self.mad = None # median absolute deviation
+        self.meanad = None # mean absolute deviation
+
+    def fit(self, x: torch.Tensor):
+        med = x.median().item()
+        self.median = med
+        self.mad = (x - med).abs().median().item()
+        self.meanad = (x - med).abs().mean().item()
+        # Having a meanad == 0 (when all values are the same), breaks training. Just use 1.0 in this case
+        if (self.meanad == 0):
+            self.meanad = 1.0
+
+    def transform(self, x: typing.Union[torch.Tensor, np.ndarray]):
+        result = ensure_float_type(x)
+
+        # reference: https://www.ibm.com/docs/el/cognos-analytics/11.1.0?topic=terms-modified-z-score
+        result -= self.median
+        if self.mad == 0:
+            result /= (1.253314 * self.meanad)
+        else:
+            result /= (1.486 * self.mad)  # 1.486 * mad approximately equals the standard deviation
```

Comment:
Change these magic numbers in favor of named variables

Reply:
Done